gap/base/methsel.gi

```diff
@@ -20,40 +20,21 @@
 # Add a method to a database with "AddMethod" and call a method from a
 # database with "CallMethods".
 #
-InstallGlobalFunction("AddMethod", function(methodDb, method)
-    local l, pos;
-    if not IsSubsetSet(NamesOfComponents(method),
-                       ["method", "rank", "stamp"]) then
-        ErrorNoReturn("<method> must have components",
-                      " method, rank, and stamp");
+InstallGlobalFunction("AddMethod", function(methodDb, method, rank)
+    local pos;
+    if not IsRecogMethod(method) then
+        ErrorNoReturn("<method> must be a RecogMethod");
     fi;
-    if not IsBound(method.comment) then
-        method.comment := "";
+    # This may be turned into an error in the future, see github issue #184.
+    if not HasValidatesOrAlwaysValidInput(method) then
```

Comment:
I just merged another PR of your where you removed this code, I think?

Reply:
I rebased it. Now it's gone here too.